src/c.mli

```diff
@@ -0,0 +1,45 @@
+open Stdune
+
+module Kind : sig
+  type t =
+    | C
+    | Cxx
+
+  val pp : t Fmt.t
+
+  val split_fn : string -> (string * t) option
+
+  val possible_fns : t -> string -> string list
+
+  module Dict : sig
+    type kind
+    type 'a t =
+      { c : 'a
+      ; cxx : 'a
+      }
+
+    val make : 'a -> 'a t
+
+    val update : 'a t -> kind -> f:('a -> 'a) -> 'a t
+
+    val merge : 'a t -> 'b t -> f:('a -> 'b -> 'c) -> 'c t
+  end with type kind := t
+end
+
+module Source : sig
+  type t
+
+  val kind : t -> Kind.t
+  val path : t -> Path.t
+  val src_dir : t -> Path.t
+
+  val make : kind:Kind.t -> path:Path.t -> t
+end
+
+module Sources : sig
+  type t = (Loc.t * Source.t) String.Map.t
+
+  val foreign_objects : t -> dir:Path.t -> ext_obj:string -> Path.t list
```

Comment:
I would just call this `objects`. They are foreign from the point of view of OCaml, but not from the point of view of this module

Reply:
done